lte/gateway/python/integ_tests/s1aptests/test_ipv4v6_non_nat_ded_bearer_dl_tcp.py

```diff
@@ -0,0 +1,188 @@
+"""
+Copyright 2022 The Magma Authors.
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import ipaddress
+import time
+import unittest
+
+import s1ap_types
+from integ_tests.s1aptests import s1ap_wrapper
+from integ_tests.s1aptests.s1ap_utils import SessionManagerUtil, SpgwUtil
+from s1ap_utils import MagmadUtil
+
+
+class TestIpv4v6NonNatDedBearerDlTcp(unittest.TestCase):
+    """Integration Test: TestIpv4v6NonNatDedBrDlTcp"""
+
+    def __init__(self, method_name: str = ...) -> None:
+        """Initialize unittest class"""
+        super().__init__(methodName=method_name)
+        self.magma_utils = MagmadUtil(None)
+        self._sessionManager_util = SessionManagerUtil()
+        self._spgw_util = SpgwUtil()
+
+    def setUp(self):
+        """Initialize before test case execution"""
+        self.magma_utils.disable_nat(ip_version=6)
+        self._s1ap_wrapper = s1ap_wrapper.TestWrapper(ip_version=6)
+
+    def tearDown(self):
+        """Cleanup after test case execution"""
+        self._s1ap_wrapper.cleanup()
+        self.magma_utils.enable_nat(ip_version=6)
+
+    def test_ipv4v6_non_nat_ded_bearer_dl_tcp(self):
+        """ipv4v6 attach with dedicated bearer and DL TCP data test with a single UE"""
+        num_ues = 1
+        magma_apn = {
+            "apn_name": "magma",  # APN-name
+            "qci": 9,  # qci
+            "priority": 15,  # priority
+            "pre_cap": 1,  # preemption-capability
+            "pre_vul": 0,  # preemption-vulnerability
+            "mbr_ul": 200000000,  # MBR UL
+            "mbr_dl": 100000000,  # MBR DL
+            "pdn_type": 2,  # PDN Type 0-IPv4,1-IPv6,2-IPv4v6
+        }
+
+        wait_for_s1 = True
+        ue_ips = ["fdee::"]
+        apn_list = [magma_apn]
+
+        self._s1ap_wrapper.configUEDevice(num_ues, [], ue_ips)
+
+        req = self._s1ap_wrapper.ue_req
+        ue_id = req.ue_id
+        print(
+            "************************* Running End to End attach for ",
+            "UE id ",
+            req.ue_id,
+        )
+
+        self._s1ap_wrapper.configAPN(
+            "IMSI" + "".join([str(j) for j in req.imsi]),
+            apn_list,
+            default=False,
+        )
+
+        # Now actually complete the attach
+        attach = self._s1ap_wrapper.s1_util.attach(
+            ue_id,
+            s1ap_types.tfwCmd.UE_END_TO_END_ATTACH_REQUEST,
+            s1ap_types.tfwCmd.UE_ATTACH_ACCEPT_IND,
+            s1ap_types.ueAttachAccept_t,
+            pdn_type=3,
```

Comment:
It would be good to mark in comment as above everywhere we are using this hardcoded pdn type value `# PDN Type 0-IPv4,1-IPv6,2-IPv4v6` (same for other testcases)